test/WebJobs.Script.Tests/Diagnostics/FileTraceWriterTests.cs

```diff
@@ -255,7 +278,7 @@ public void GetTracePrefix_ReturnsExpectedValue()
         [Fact]
         public async Task Trace_LogFileDeleted_CreatesNewFile()
         {
-            var traceWriter = new FileTraceWriter(_logFilePath, TraceLevel.Info, LogType.Host);
+            var traceWriter = new TestFileTraceWriter(_logFilePath, TraceLevel.Info, LogType.Host, true);
```

Comment:
Are we still testing the actual product now? It seems like you're overriding behavior now so tests behave differently than product?

Reply:
I debated modifying those tests, but this is essentially replicating the behavior we'll have in production with the scheduled flushes. The problem is that these tests expect the logs to be flushed immediately after that call, but they may be delayed until the timer fires.